src/js/components/Carousel/Carousel.js

```diff
@@ -1,242 +1,222 @@
-import React, {
-  Children,
-  useCallback,
-  useContext,
-  useEffect,
-  useState,
-  useRef,
-} from 'react';
+import React, { Children, useRef, useState, useEffect } from 'react';
+import { CarouselChild } from './CarouselChild';
+import { CarouselControls } from './CarouselControls';
+import { StyledCarouselContainer } from './StyledCarousel';
+import { CarouselPropTypes } from './propTypes';
+import { Keyboard } from '../Keyboard';
 
-import { normalizeColor } from '../../utils';
-import { defaultProps } from '../../default-props';
-import { ThemeContext } from '../../contexts';
+const handleOnJumpNavigation =
+  ({ setDirection, setCurrent, setInTransition, setPrevious, onChild }) =>
+  (current, index, inTransition) => {
+    if (current === index) return;
+    if (inTransition) return;
+    setDirection(index > current ? 'next' : 'previous');
+    setInTransition(true);
+    setPrevious(current);
+    setCurrent(index);
+    onChild(index);
+  };
 
-import { Box } from '../Box';
-import { Button } from '../Button';
-import { Keyboard } from '../Keyboard';
-import { Stack } from '../Stack';
+const handleOnNext =
+  ({
+    setDirection,
+    setCurrent,
+    setPrevious,
+    setInTransition,
+    numSlides,
+    onChild,
+  }) =>
+  (current, inTransition) => {
+    if (inTransition) return;
+    setDirection('next');
+    setInTransition(true);
+    setPrevious(current);
+    const next = current === numSlides - 1 ? 0 : current + 1;
+    setCurrent(next);
+    onChild(next);
+  };
 
-import { CarouselChild } from './CarouselChild';
-import { CarouselPropTypes } from './propTypes';
+const handleOnPrevious =
+  ({
+    setDirection,
+    setCurrent,
+    setPrevious,
+    setInTransition,
+    numSlides,
+    onChild,
+  }) =>
+  (current, inTransition) => {
+    if (inTransition) return;
+    setDirection('previous');
+    setInTransition(true);
+    setPrevious(current);
+    const next = current === 0 ? numSlides - 1 : current - 1;
+    setCurrent(next);
+    onChild(next);
+  };
 
 const Carousel = ({
   activeChild,
-  initialChild,
-  onChild,
-  play,
   children,
   controls,
+  height,
   fill,
-  onFocus,
-  onBlur,
-  ...rest
+  width,
+  initialChild,
+  onChild,
+  play,
 }) => {
-  const theme = useContext(ThemeContext) || defaultProps.theme;
-  const [focus, setFocus] = useState();
-  const timerRef = useRef();
-
-  const [indexes, setIndexes] = useState({
-    activeIndex: activeChild !== undefined ? activeChild : initialChild,
+  const noContainer = !fill && !height && !width;
+  const numSlides = children.length;
+  const firstChildRef = useRef(null);
+  const [current, setCurrent] = useState(initialChild);
+  const [previous, setPrevious] = useState(undefined);
+  const [containerProps, setContainerProps] = useState({
+    heightProp: height,
+    widthProp: width,
```

Comment:
I think we can get away with:
```suggestion
    height,
    width,
```

If more readable, we tend to rename the destructured prop being passed by the caller to [variable_name]Prop to make it clear where the value is originating from. This is more of a convention.

```suggestion
    height: heightProp,
    width: widthProp,
```

Reply:
I've seen in the past that when I spread those props of `height` and `width` to the Styled Box those values are read by the Box and overwrite what is happening the in the Styled Box css logic. But gonna look into this again.

Reply:
Same comment above about styled component